test/Microsoft.ML.CodeAnalyzer.Tests/TypeIsSchemaShapeTest.cs

```diff
@@ -2,94 +2,89 @@
+        public async Task ReturnTypeIsSchemaShapeClass()
         {
             // This is a somewhat more complex example, where instead of direct usage the user of the API is devising their own
             // function where the shape type is a generic type parameter. In this case, we would ideally like the analysis to get
             // chained out of their function.
-            var analyzer = GetCSharpDiagnosticAnalyzer();
-            var diag = analyzer.SupportedDiagnostics[0];
-            var diagTp = analyzer.SupportedDiagnostics[1];
-            var diagC = analyzer.SupportedDiagnostics[2];
-            var diagF = analyzer.SupportedDiagnostics[3];
-
-            var diagPg = analyzer.SupportedDiagnostics[4];
-            var diagPnas = analyzer.SupportedDiagnostics[5];
-            var diagPs = analyzer.SupportedDiagnostics[6];
-            var diagPc = analyzer.SupportedDiagnostics[7];
-
             var expected = new DiagnosticResult[] {
-                diagF.CreateDiagnosticResult(33, 13, "Class4", "F1"),
-                diagC.CreateDiagnosticResult(34, 13, "Class5"),
-                diagC.CreateDiagnosticResult(35, 13, "Class6"),
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeClassDiagnosticField.Rule).WithLocation(33, 13).WithArguments("Class4", "F1"),
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeClassDiagnosticConstructor.Rule).WithLocation(34, 13).WithArguments("Class5"),
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeClassDiagnosticConstructor.Rule).WithLocation(35, 13).WithArguments("Class6"),
+
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeClassDiagnosticNoArgsSettable.Rule).WithLocation(36, 13).WithArguments("Class7", "F1"),
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeClassDiagnosticArgsSettable.Rule).WithLocation(37, 13).WithArguments("Class8", "F2"),
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeClassDiagnosticGettable.Rule).WithLocation(38, 13).WithArguments("Class9", "F2"),
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeClassDiagnosticCorrespondence.Rule).WithLocation(39, 13).WithArguments("Class10"),
+                VerifyCS.Diagnostic(TypeIsSchemaShapeAnalyzer.ShapeClassDiagnosticCorrespondence.Rule).WithLocation(40, 13).WithArguments("Class11"),
 
-                diagPnas.CreateDiagnosticResult(36, 13, "Class7", "F1"),
-                diagPs.CreateDiagnosticResult(37, 13, "Class8", "F2"),
-                diagPg.CreateDiagnosticResult(38, 13, "Class9", "F2"),
-                diagPc.CreateDiagnosticResult(39, 13, "Class10"),
-                diagPc.CreateDiagnosticResult(40, 13, "Class11"),
+                new DiagnosticResult("CS0246", DiagnosticSeverity.Error).WithLocation(44, 71).WithMessage("The type or namespace name 'MissingClass' could not be found (are you missing a using directive or an assembly reference?)"),
```

Comment:
📝 Based on the name, it seems this could be intentional. If it's not, let me know.